lib/date_time/date_time.c

```diff
@@ -631,19 +92,18 @@ int date_time_uptime_to_unix_time_ms(int64_t *uptime)
 
 	uptime_prev = *uptime;
 
-	if (!initial_valid_time) {
-		LOG_WRN("Valid time not currently available");
-		return -ENODATA;
+	err = date_time_now(&date_time_ms);
+	if (err) {
+		return err;
 	}
 
-	*uptime += time_aux.date_time_utc - time_aux.last_date_time_update;
+	*uptime += date_time_ms - date_time_core_last_update_uptime();
```

Comment:
Moving this functionality to `date_time_core.c` would mean `date_time_core_last_update_uptime()` is only needed within that module and not in its API. Drawback is that it would be another case where `date_time.c` forwards the call to `date_time_core.c`.

Reply:
Leaving this to `date_time.c`